chef/cookbooks/ceph/recipes/osd.rb

```diff
@@ -68,7 +68,8 @@
 
   if is_crowbar?
     node.set["ceph"]["osd_devices"] = [] if node["ceph"]["osd_devices"].nil?
-    unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node).sort
+    min_size_blocks = node["ceph"]["osd"]["min_size_gb"] * 1024 * 1024 * 2
+    unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node).sort.select {|d| d.size >= min_size_blocks}
```

Comment:
Yeah, seems to work fine (i.e. it returns the disk size in 512 byte blocks). How long ago was it broken? AFAICT there don't seem to have been any changes the relevant code in `ohai/files/default/plugins/linux/block_device_extended.rb` or `barclamp/libraries/barclamp_library.rb` for rather a long time.